src/recipes.jl

```diff
@@ -0,0 +1,85 @@
+"""
+    Revise.track(Base)
+
+Track the code in Julia's `base` directory for updates. This
+facilitates making changes to Julia itself and testing them
+immediately (without rebuilding).
+
+At present some files in Base are not trackable, see the README.
+"""
+function track(mod::Module)
+    if mod == Base
+        # Determine when the basesrccache was built
+        mtcache = mtime(basesrccache)
+        # Initialize expression-tracking for files, and
+        # note any modified since Base was built
+        files = String[]
+        for (submod, filename) in Base._included_files
+            submod == Main || startswith(String(nameof(submod)), "Base") || continue
+            push!(file2modules, filename=>FileModules(submod, ModDict(), basesrccache))
+            push!(files, filename)
+            if mtime(filename) > mtcache
+                push!(revision_queue, filename)
+            end
+        end
+        # Add the files to the watch list
+        init_watching(files)
+    elseif mod == Core.Compiler
+        compilerdir = joinpath(juliadir, "base", "compiler")
+        track_subdir_from_git(Core.Compiler, compilerdir)
+    elseif nameof(mod) ∈ stdlib_names
+        stdlibdir = joinpath(juliadir, "stdlib")
+        libdir = joinpath(stdlibdir, String(nameof(mod)), "src")
+        track_subdir_from_git(mod, normpath(libdir))
+    else
+        error("no Revise.track recipe for module ", mod)
+    end
+    nothing
+end
+
+# For tracking subdirectories of Julia itself (base/compiler, stdlibs)
+function track_subdir_from_git(mod::Module, subdir::AbstractString)
+    # diff against files at the same commit used to build Julia
+    repo, repo_path = git_repo(subdir)
+    if repo == nothing
+        error("could not find git repository at $subdir")
+    end
+    prefix = subdir[length(repo_path)+2:end]   # git-relative path of this subdir
+    tree = git_tree(repo, Base.GIT_VERSION_INFO.commit)
+    files = Iterators.filter(file->startswith(file, prefix), keys(tree))
+    ccall((:giterr_clear, :libgit2), Cvoid, ())  # necessary to avoid errors like "the global/xdg file 'attributes' doesn't exist: No such file or directory"
```

Comment:
I'll look into this more but I think this is pointing to a LibGit2 issue.

Reply:
Details on the issue have been filed with [libgit2](https://github.com/libgit2/libgit2/issues/4693). As a fix for the Julia LibGit2 library I've proposed https://github.com/JuliaLang/julia/pull/27763 which should cleanly work around the issue. For now your `giterr_clear` solution will work just fine.